en/05_Contributing/05_Coding_Conventions/01_PHP_Coding_Conventions.md

```diff
@@ -136,6 +136,133 @@ class MyClass extends Class
+
+## Directory naming - src dir
+
+- Put similar types of classes in the same directory. For example, put Controllers in a Controllers directory
+- Put DataObjects in a directory called Models
+- Do not group different class types together in a directory relating to a single feature
+- Use initial caps pluralised for the directory naming e.g. use `Controllers` rather than `Controller` or `controllers`
+
+## Properties and accessors
+
+- Do not use public properites, instead use a combination of private properites and public getters and setters
+- For setters use a combination of a `:static` return type and `return $this` to allow for fluent programming
+- Do not use [dynamic properties](https://www.php.net/manual/en/language.oop5.properties.php#language.oop5.properties.dynamic-properties)
+
+## Method and property visibility
+
+- Use private over protected as this reduces the maintained API surface. Only use protected if there's an immediate reason to upgrade from private.
+- Note that there is a lot of legacy Silverstripe code where protected was used over private.
+
+## Interfaces
+
+- If there are or there's a realistic expection that there will be multiple class implementations of something then create an interface and use that for parameter and return types, and as the service key for injector.
+- If there is only a single implemention and no forseeable expectation of more implementaitons being added, then do not create an interface for the single class implementation because this would only be adding unecessarily complexity.
+
+## Typing
+
+- Aim to strongly type all new properties, method parameters, and method return types.
+- The exception to this is adding a new method or property to a class which is entirely untyped and adding typed API would be very inconsistent with the rest of the file. In this instance use PHP docblocks to define types.
+- If there are duplicate strong types and docblock types then remove the old docblock types.
+- Do not strongly type existing API outside of a new major version as it's an API breakage.
+
+## Mixed types
+
+- Where possible avoid using mixed types, including type unions, because this leads to complicated and brittle code.
+- Try to avoid using nullable scalars and instead use the default value of the scalar instead of null. One exception to this is for when `null` means 'uninitialised`.
+- Avoid the use of the splat `...` parameter as this makes API enforcement difficult. If you need to support multiple arguments then use an array parameter type.
+
+## Object instantiation
+
+- Use Injector to instantiate objects rather than the `new` keyword. This is done as it allows projects to swap out classes with different implementations.
+- Note there is only a minimal performance penalty for this, though there is a downside that IDE type hinting stops working. During development you may wish to use the new keyword for this reason and change it to use injector before submitting your PR.
```

Comment:
```suggestion
```
I don't think this is necessarily useful to say - _maybe_ we can keep the part about performance, but that's not really a standard so probably belongs outside the list.
The part about what people do during development doesn't belong here IMO.

Reply:
Removed